examples/SLAM/js/hit-test-location-xr8.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, Object as WLEObject, Type } from '@wonderlandengine/api';
+import { vec3 } from 'gl-matrix';
+import { ARSession, XR8Provider, ARProvider } from '../../../';
+
+import { Plane } from './common/math/plane';
+import { Ray } from './common/math/ray';
+
+/**
+ * Hit test location for XR8 implementation
+ */
+class HitTestLocationXR8 extends Component {
+  public static TypeName = 'hit-test-location-xr8';
+  public static Properties = {
+    camera: { type: Type.Object },
+  };
+  private tempScaling = new Float32Array(3);
+  // injected by WLE
+  camera!: WLEObject;
+
+  private tracking = false;
+  private glPlane: Plane = new Plane(vec3.fromValues(0, 1, 0), 0);
+
+
+  private camForward = new Array(3);
+  private camForwardVec3 = vec3.create();
+  private intersectionVec3 = vec3.create();
+  private tmpWorldPosition = new Array(3);
```

Comment:
Generally, gl-matrix supports array as param as well, btw, so no need to convert. I recommend `Float32Array` for minimally better performance, that's what .create() does, I think.